redbot/cogs/streams/streamtypes.py

```diff
@@ -124,6 +134,7 @@ def __init__(self, **kwargs):
         # self.already_online = kwargs.pop("already_online", False)
         self._messages_cache = kwargs.pop("_messages_cache", [])
         self.type = self.__class__.__name__
+        self.token_name = None
```

Comment:
Here you can remove token_name as an instance variable, but keep it as a class variable annotation:
```py
from typing import ClassVar

...

class Stream:

    token_name: ClassVar[str]

    def __init__(self, **kwargs):
```